File: Codes in Python3/Graph/Clone Graph.py
from os import *
from sys import *
from collections import *
from math import *

# Class for graph node is as follows:
class graphNode:
    def __init__(self, *args):
        if(len(args) == 0):
            self.data = 0
            self.neighbours = []

        elif(len(args) == 1):
            self.data = args[0]
            self.neighbours = []

        elif(len(args) == 2):
            self.data = args[0]
            self.neighbours = args[1]

    def __del__(self):
        self.neighbours.clear()


def cloneGraph(node):
    oldToNew={}
    def dfs(node):
        if node in oldToNew:
            return oldToNew[node]
        copy=graphNode(node.data)
        oldToNew[node]=copy
        for nei in node.neighbours:
            copy.neighbours.append(dfs(nei))
        return copy

    return dfs(node) if node else None
    pass

File: Codes in C++/Graph/Clone Graph.cpp
#include <bits/stdc++.h> 
/***************************************************************************

	Class for graph node is as follows:

	class graphNode
	{
		public:
    		int data;
    	vector<graphNode *> neighbours;
    	graphNode()
    	{
        	data = 0;
        	neighbours = vector<graphNode *>();
    	}

    	graphNode(int val)
    	{
        	data = val;
        	neighbours = vector<graphNode *>();
    	}

    	graphNode(int val, vector<graphNode *> neighbours)
    	{
        	data = val;
        	this->neighbours = neighbours;
    	}
	};

******************************************************************************/
graphNode *dfs(graphNode *node,map<graphNode*,graphNode*> &oldToNew)
{
    if(oldToNew.find(node)!=oldToNew.end())
        return oldToNew[node];
    graphNode* copy= new graphNode(node->data);
    oldToNew[node]=copy;
    for(auto nei:node->neighbours)
        copy->neighbours.push_back(dfs(nei,oldToNew));
    return copy;
}

graphNode *cloneGraph(graphNode *node)
{
    if(node==nullptr)
     return nullptr;
    map<graphNode*,graphNode*> oldToNew;
    return dfs(node,oldToNew);
}